nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/logging/repository/StandardLogRepository.java

```diff
@@ -27,14 +29,16 @@
 
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.EnumMap;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Optional;
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 public class StandardLogRepository implements LogRepository {
 
-    private final Map<LogLevel, Collection<LogObserver>> observers = new HashMap<>();
+    private final EnumMap<LogLevel, Collection<LogObserver>> observers = new EnumMap<>(LogLevel.class);
```

Comment:
Good idea to use an `EnumMap` here instead of a `HashMap`. Should use a declaration of `Map`, though.